scripts/sanity_chk/sanitylib.py

```diff
@@ -3229,6 +3232,45 @@ def csv_report(self, filename):
                     rowdict["rom_size"] = rom_size
                 cw.writerow(rowdict)
 
+    def set_default(self, obj):
+        ''' JSON doesn't recognize set type so need to converted to list '''
+        if isinstance(obj,set):
+            return list (obj)
+        raise TypeError
+
+    def json_report(self, filename):
+        rowdict = {}
+        rowdict["environment"] = {"OS": os.name,
+                                    "Toolchain": self.get_toolchain()
+                                    }
+        rowdict["testspec"] = []
+        for instance in self.instances.values():
+            rowdict["testspec"].append({"test": instance.testcase.name,
+                                        "arch": instance.platform.arch,
+                                        "platform": instance.platform.name,
+                                        "handler": instance.platform.simulation,
+                                        "status": instance.status,
+                                        "extra_args": instance.testcase.extra_args,
+                                        "extra_configs": instance.testcase.extra_configs,
+                                        "harness": instance.testcase.harness,
+                                        "harness_config": instance.testcase.harness_config,
+                                        "type": instance.testcase.type,
+                                        "tags": instance.testcase.tags
+                                        })
+            if instance.status not in ["failed", "timeout"]:
+                if instance.handler:
+                    rowdict["testspec"].append({"handler_time": (instance.metrics.get           ("handler_time", 0))})
+                rowdict["testspec"].append({"ram_size": instance.metrics.get("ram_size", 0),
+                                            "rom_size": instance.metrics.get("rom_size",0)
+                                            })
+        rowdict["apply_filters"] = {"discards": str(self.apply_filters())}
+        with open(os.path.join(ZEPHYR_BASE, "sanity-out", "sanitycheck.log"), "r") as log_file:
```

Comment:
This is attaching the sanitycheck log which basically had everything in verbose mode from running the command, not sure how this is useful, we have the file already, the logs should be per instance(build.log, handler.log and device.log when something went wrong, I would have handler.log all the time, even in case of success.

Reply:
per instance it adds the handler and build logs

Reply:
missed adding device logs, working on adding it.

Reply:
Added device logs when the tests fails or errors out.